db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -2677,6 +2682,8 @@ Status DBImpl::WaitUntilFlushWouldNotStallWrites(ColumnFamilyData* cfd,
       // mode due to pending compaction bytes, but that's less common
       // No extra immutable Memtable will be created if the current Memtable is
       // empty.
+      // NOTE: If this ever changes, check the write stall condition criteria
+      // in `ShouldRescheduleFlushRequestToRetainUDT` and make them in sync.
```

Comment:
Maybe add a common function like `bool WouldFlushStallWrites(...)` and have them both use that.

Reply:
That's a good idea. I added a `ColumnFamilyData::GetUnflushedMemTableCountForWriteStallCheck` to be shared.  I didn't encapsulate all the write stall checking logic to be shared because logic related to L0 files and bytes needed compaction are not exactly the same.  This rescheduling check doesn't involve those two.